src/psyclone/tests/psyir/transformations/intrinsics/sum2code_trans_test.py

```diff
@@ -0,0 +1,504 @@
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Authors R. W. Ford, STFC Daresbury Laboratory
+
+'''Module containing tests for the sum2code transformation.'''
+
+import pytest
+
+from psyclone.psyir.frontend.fortran import FortranReader
+from psyclone.psyir.backend.fortran import FortranWriter
+from psyclone.psyir.nodes import BinaryOperation, Literal
+from psyclone.psyir.symbols import REAL_TYPE
+from psyclone.psyir.transformations import Sum2CodeTrans, TransformationError
+
+
+def test_initialise():
+    ''' Test that we can create an instance of the transformation '''
+    trans = Sum2CodeTrans()
+    assert isinstance(trans, Sum2CodeTrans)
+    assert (str(trans) == "Convert the PSyIR SUM intrinsic to equivalent "
+            "PSyIR code.")
+    assert trans.name == "Sum2CodeTrans"
+
+
+def test_validate_node():
+    '''Check that an incorrect node raises the expected exception.'''
+    trans = Sum2CodeTrans()
+    with pytest.raises(TransformationError) as info:
+        trans.validate(None)
+    assert ("Error in Sum2CodeTrans transformation. The supplied node "
+            "argument is not a SUM operator, found 'NoneType'."
+            in str(info.value))
+
+    bin_op = BinaryOperation.create(
+        BinaryOperation.Operator.MUL, Literal("1.0", REAL_TYPE),
+        Literal("1.0", REAL_TYPE))
+    with pytest.raises(TransformationError) as info:
+        trans.validate(bin_op)
+    assert ("Error in Sum2CodeTrans transformation. The supplied node "
+            "operator is invalid, found 'Operator.MUL'." in str(info.value))
```

Comment:
Should this message say what was expected?

Reply:
Added expected operators in error message.